xbmc/cores/VideoPlayer/DVDCodecs/Overlay/contrib/cc_decoder.c

```diff
@@ -87,26 +78,52 @@ static int good_parity(uint16_t data)
   return ret;
 }
 
-static void build_char_table(void)
+// @todo add support to CCSET_EXTENDED_SPANISH_FRENCH_MISC
+// and CCSET_EXTENDED_PORTUGUESE_GERMAN_DANISH
+enum cc_charset
 {
-  int i;
-  /* first the normal ASCII codes */
-  for (i = 0; i < 128; i++)
-   chartbl[i] = (char) i;
- /* now the special codes */
- chartbl[0x2a] = 0xA1;
- chartbl[0x5c] = 0xA9;
- chartbl[0x5e] = 0xAD;
- chartbl[0x5f] = 0xB3;
- chartbl[0x60] = 0xAA;
- chartbl[0x7b] = 0xA7;
- chartbl[0x7c] = 0xb7;
- chartbl[0x7d] = 0x91;
- chartbl[0x7e] = 0xB1;
- chartbl[0x7f] = 0xA4;    /* FIXME: this should be a solid block */
-}
-
-static void ccbuf_add_char(cc_buffer_t *buf, uint8_t c)
+  CCSET_BASIC_AMERICAN,
+  CCSET_SPECIAL_AMERICAN,
+};
+
+// from ffmpeg EIA-608 decoder
+// see https://en.wikipedia.org/wiki/EIA-608#Characters for reference
+static const char* charset_overrides[2][128] = {
+    [CCSET_BASIC_AMERICAN] =
+        {// The code is almost identical to ASCII apart from these exceptions
+         [0x27] = "\u2019",
+         [0x2a] = "\u00e1",
+         [0x5c] = "\u00e9",
+         [0x5e] = "\u00ed",
+         [0x5f] = "\u00f3",
+         [0x60] = "\u00fa",
+         [0x7b] = "\u00e7",
+         [0x7c] = "\u00f7",
+         [0x7d] = "\u00d1",
+         [0x7e] = "\u00f1",
+         [0x7f] = "\u2588"},
```

Comment:
is there a missing } somewhere in this block?

Reply:
Nop, the closing } is at the end of this line

Reply:
Ah yeah, maybe that's what's causing clang to do weird things, maybe bump the }, to the next line maybe